src/Microsoft.SqlTools.ServiceLayer/TableDesigner/Contracts/DataModel/TableDataModel.cs

```diff
@@ -0,0 +1,44 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using Newtonsoft.Json;
+
+namespace Microsoft.SqlTools.ServiceLayer.TableDesigner.Contracts
+{
+    /// <summary>
+    /// The data model for a table object
+    /// </summary>
+    public class TableDataModel : ObjectDataModelBase
+    {
+        public TableDataModel()
+        {
+            this.Columns = new TableColumnCollection();
+            this.Schema = new DropdownProperties();
+            this.Description = new InputBoxProperties();
+            this.Script = new InputBoxProperties();
+        }
+        public DropdownProperties Schema { get; set; }
+
+        public InputBoxProperties Description { get; set; }
+
+        public TableColumnCollection Columns { get; set; }
+
+        public InputBoxProperties Script { get; set; }
+    }
+
+    public class TableColumnCollection : TableProperties<TableColumnDataModel>
+    {
+        [JsonIgnore]
+        protected override string NewObjectNamePrefix { get { return "column"; } }
+
+        protected override TableColumnDataModel CreateNew(string name)
+        {
+            //TODO: Add the default values
+            var column = new TableColumnDataModel();
+            column.Name.Value = this.GetDefaultNewObjectName();
```

Comment:
The fact that this is tying the model to the specific type of UI to use doesn't seem like something we normally want. Why isn't the model just the current values for the object that gets updated when a user selects a new one?